nncf/experimental/openvino_native/engine.py

```diff
@@ -36,19 +36,21 @@ def __init__(self, model: ov.Model, target_device: TargetDevice = TargetDevice.C
         self.compiled_model = ie.compile_model(model, target_device.value)
         self.input_names = set(inp.get_friendly_name() for inp in model.get_parameters())
 
-    def infer(self, input_data: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+    def infer(self, input_data: Union[np.ndarray, List[np.ndarray], Tuple[np.ndarray], Dict[str, np.ndarray]]) \
+              -> Dict[str, np.ndarray]:
         """
         Runs model on the provided input via OpenVINO Runtime.
         Returns the dictionary of model outputs by node names.
 
         :param input_data: inputs for the model.
         :return output_data: models outputs.
         """
-        model_inputs = {}
-        for name in self.input_names:
-            if name not in input_data:
-                raise RuntimeError(f'Missing a required input: {name} to run the model.')
-            model_inputs[name] = input_data[name]
-
-        model_outputs = self.compiled_model(model_inputs)
+        if len(self.input_names) != len(input_data):
```

Comment:
Does it correctly works when `input_data` is an `np.ndarray`?

Reply:
You are right. For `np.ndarray` len returns `np.ndarray.shape[0]`. Thanks for comment!